Assets/Scripts/HUD/Inventory.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Collections.Generic;
+using MonoBehaviours.WeaponsSystem;
+using Player;
+using UnityEngine;
+
+namespace HUD
+{
+    public class Inventory : MonoBehaviour
+    {
+        /// <summary>
+        /// Dictionary containing the inventory slots of the HUD inventory. Keys are slots, and values are the
+        /// game objects displaying the HUD.
+        /// </summary>
+        private Dictionary<InventoryIndex, GameObject> _inventoryContainer;
+        private InventoryIndex _selectedSlot;
+
+        void Start()
+        {
+            _inventoryContainer = new Dictionary<InventoryIndex, GameObject>
+            {
+                {InventoryIndex.First, transform.Find("Weapon1").gameObject},
+                {InventoryIndex.Second, transform.Find("Weapon2").gameObject},
+                {InventoryIndex.Throwable, transform.Find("Throwable").gameObject}
+            };
+        }
+
+        //Outlines the weapon image in the HUD inventory that is currently used by the player.
+        public void ChangeSelected(InventoryIndex newSelectedSlot)
+        {
+            _inventoryContainer[_selectedSlot].transform.Find("Selected").gameObject.SetActive(false);
+            _selectedSlot = newSelectedSlot;
+            _inventoryContainer[_selectedSlot].transform.Find("Selected").gameObject.SetActive(true);
+        }
+
+        public void RemoveFromInventory(InventoryIndex weaponSlot)
+        {
+            _inventoryContainer[weaponSlot].SendMessage("RemoveWeaponFromInventorySlot", SendMessageOptions.DontRequireReceiver);
+        }
+
+        public void AddToInventory(InventoryIndex slot, WeaponBehaviourScript weaponScript)
+        {
+            _inventoryContainer[slot].SendMessage("AddWeaponToInventorySlot", weaponScript, SendMessageOptions.DontRequireReceiver);
```

Comment:
This'll cause an issue if the player pickups a weapon they already have. The gameobject that the weaponScript is attached to is destroyed right after it's added to the inventory (since we only want to add the ammo to the weapon already in the inventory)

Reply:
You're right. In that case, you don't need to fire an event because the weapon is already in the inventory and the ammo will be updated by itself. I'll remove the event for that case inside the inventory manager.